samples/corda/corda-simple-application/workflows-kotlin/src/main/kotlin/com/example/flow/AssetFlow.kt

```diff
@@ -624,3 +632,204 @@ class AcceptorOfTransfer(val otherPartySession: FlowSession) : FlowLogic<SignedT
     }
 }
 
+/**
+ * The GetAssetJsonStringFromStatePointer flow fetches the asset state from its state pointer [AssetPledgeState].assetStatePointer
+ * and creates marshalled JSON encoded object which is returned.
+ * This function is called by the exporting network in the context of interop-query from the importing network
+ * to the exporting network before performing claim on remote asset.
+ *
+ * @property assetPledgeState The (interop) vault state that represents the pledge details on the ledger.
+ */
+@InitiatingFlow
+@StartableByRPC
+class GetAssetJsonStringFromStatePointer(
+    val assetPledgeState: AssetPledgeState
+) : FlowLogic<String>() {
+    @Suspendable
+    override fun call(): String {
+
+        if (assetPledgeState.assetStatePointer == null) {
+            // Typically, [AssetPledgeState].assetStatePointer will be null only in the case of pledge details not
+            // being available for a given pledgeId. The flow GetAssetPledgeStatus in AssetTransferFlows sets this
+            // pointer to null if the pledge-state is not available in the context of the interop-query from the
+            // importing n/w to the exporting n/w. Hence return empty string, and this will not be passed to the
+            // JSON unmarshalling method GetTokenStateAndContractId since the expiryTime will be elapsed for the
+            // claim to happen (i.e., if assetStatePointer is null, then expiryTimeSecs will be set to past time).
+            return ""
+        }
+
+        val assetStatePointer: StaticPointer<ContractState> = assetPledgeState.assetStatePointer!!
+        val assetState = assetStatePointer.resolve(serviceHub).state.data as AssetState
+
+        println("Creating simple asset JSON from StatePointer.")
+        var marshalledAssetJson =
+            subFlow(MarshalFungibleAsset(assetState.tokenType, assetState.quantity, assetPledgeState.lockerCert))
```

Comment:
subFlow calls are expensive, like chaincode to chaincode calls in fabric, I think it should be avoided when possible? If you agree, then it seems that this subFlow is called only here, so may be move its contents directly here?

Reply:
Changed subFlow to function now.